pkg/operator/csidriveroperator/crcontroller.go

```diff
@@ -240,6 +221,61 @@ func (c *CSIDriverOperatorCRController) applyClusterCSIDriver(requiredOriginal *
 	return actual, true, err
 }
 
+func (c *CSIDriverOperatorCRController) syncConditions(conditions []operatorapi.OperatorCondition, updatefn v1helpers.UpdateStatusFunc) error {
```

Comment:
Do you think it would be more readable to split this into 2 functions: one to handle optional operators and the other to handle non-optional ones? Or have a big if-else clause (if c.optional)?

Reply:
I tried and it looks even worse - two functions that look like copies, with only small differences.